shor/gates.py

```diff
@@ -518,28 +1489,110 @@ def to_matrix(self) -> np.ndarray:
 
 
 class Cr(_Gate):
+    """
+    Apply the control phase shift gate Cr or sometimes referred as CU1. First qubit being the control and second target qubit
+    has a phase shift applied.
+    Inherits from gate class
+
+    Attributes
+    __________
+    symbol : str
+        a string used to represent gate for provider transpiler
+    qubits : int
+        qubit index to which the Cr gate is applied
+    angle : float
+        Floating point number containing angle of phase shift
+    dimension : int
+        number of qubits Cr gate is applied to
+
+    Methods
+    -------
+    to_matrix(self) -> np.ndarray
+        Returns matrix form of gate as numpy array
+    """
+
     symbol = "CU1"
 
     def __init__(self, *qubits, angle, **kwargs):
+        """
+        Parameters
+        __________
+        qubits : int
+            qubit indexes apply Cr gate. First is a control
+            qubit and second is target qubit.
+        angle : float
+            anlge used to rotae qubit of choice around the z-axis
+        kwargs : keyword str containing value int
+            keyword - dimension containing value int 2**n where n is number of qubits
+        """
         self.angle = angle
         kwargs["dimension"] = 2
         if not qubits:
             qubits = [0]
         super().__init__(*qubits, **kwargs)
 
     def to_matrix(self) -> np.ndarray:
+        """Returns matrix form of gate as numpy array
+        Needs no inputs
+
+        Returns
+        -------
+        numpy array
+            matrix form of gate with the size 2x2
+        """
         return np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, np.exp(1j * self.angle)]], dtype="complex")
 
 
 class CRk(_Gate):
+    """
+    Apply the parameteruzed control phase shift gate Crk with angle pi/2**k. First qubit being the control and second target qubit
+    has a phase shift applied.
+    Inherits from gate class
```

Comment:
I think inheritance gets picked up automatically.. Can probably get rid of these.

Reply:
Okay didnt know, should be a pretty easy fix in the future